### DISCUSSION TITLE
Merge this project with the official Spring Pulsar

### DISCUSSION THREAD

**Discussion by migroskub** [majusko/pulsar-java-spring-boot-starter#301] [General]
Now that there is an official Spring Pulsar project - https://github.com/spring-projects-experimental/spring-pulsar, I suggest that instead of keeping this project as a separate Spring Pulsar wrapper, you can try to contribute features that have been implemented here to the official project. Same as for open issues that exist here (which are probably relevant there too).
Most of the logic that has been implemented here can be used there, and this way the official project can get improved immediately. The advantages are that in the official project there are good maintainers that can easily extend any feature that will get contributed to them.

**Comment by oguzhantortop** [comment]
Is this project driven by Spring framework?

**Comment by majusko** [comment]
Hey @migroskub and @oguzhantortop , this is a community-based project and not a project maintained by a private company.

The project is here for a while and therefore is widely used and has many downloads. We cannot just simply stop working on such a project :) It has great test coverage and is used in many production environments and projects that run millions of daily users. 

It's normal that there are multiple libraries for the same thing. Feel free to contribute to it, it's much more probable that your PR will be accepted than in a library that is supported by a corporation such as VMWare.

**Reply by oguzhantortop** [reply]
Actually, I asked about https://github.com/spring-projects-experimental/spring-pulsar this project. I also think that there should be several solutions addressing the same functionality...

**Reply by majusko** [reply]
Aah, that makes sense now 🫣😂 ... 👍
